docs/MPURegionsUsageGuide.md

```diff
@@ -31,39 +31,43 @@
 1. **How to add new MPU region**
-	- Logical mapping of MPU regions to their respective memory segments is done statically using below enumeration in common mpu.h header file (os/include/tinyara/mpu.h).
+	- Logical mapping of MPU regions to their respective memory segments is done statically using below structure in common mpu.h header file (os/include/tinyara/mpu.h).
 	```c
-	enum MPU_REG_NUM {
-	#ifdef CONFIG_APP_BINARY_SEPARATION
-	#ifdef CONFIG_SUPPORT_COMMON_BINARY
-	#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
-		MPU_REG_NUM_COM_LIB_TXT,	/* Common Binary */
-		MPU_REG_NUM_COM_LIB_RO,
-		MPU_REG_NUM_COM_LIB_DATA,
-	#else
-		MPU_REG_NUM_COM_LIB,
-	#endif
-	#endif
-	#ifdef CONFIG_OPTIMIZE_APP_RELOAD_TIME
-		MPU_REG_NUM_APP_TXT,		/* Apps */
-		MPU_REG_NUM_APP_RO,
-		MPU_REG_NUM_APP_DATA,
-	#else
-		MPU_REG_NUM_APP,
-	#endif
-	#ifdef CONFIG_MPU_STACK_OVERFLOW_PROTECTION
-		MPU_REG_NUM_STK,		/* Stack */
-	#endif
-	#else
-		MPU_REG_NUM_UFLASH,
-		MPU_REG_NUM_URAM,
-	#endif
-		MPU_REG_NUM_MAX
+	struct mpu_usages_s {
+	        uint8_t nregion_board_specific;
+	        uint8_t nregion_common_bin;
+	        uint8_t nregion_app_bin;
+	        uint8_t nregion_stackovf;
+	        uint8_t max_nregion;
 	};
 	```
-	- The ordering of MPU regions in this enum is kept as above taking into consideration below rule:
+	- These regions are initialized to their respective numbers during MPU initialization in mpuinit.c file (arch/arm/src/common/up_mpuinit.c) as below, where, MPU_NUM_REGIONS is set to:  
+		- 3 for three separate MPU regions (text, ro and rw) to optimize reloading time, or
+		- 1 for just one MPU region for all of section data.
+	```c
+	void mpu_region_initialize(struct mpu_usages_s *mpu)
+
+	offset += MPU_NUM_REGIONS;
+	mpu->nregion_xxx_xx = offset;
+	```
+	- If a new type of MPU usage is to be added, then another entry has to be done in the below enum in common mpu.h header file (os/include/tinyara/mpu.h).
+	```c
+	enum mpu_region_usages_e {
+	        MPU_REGION_BOARD_SPECIFIC,
+	        MPU_REGION_COMMON_BIN,
+	        MPU_REGION_APP_BIN,
+	        MPU_REGION_STACKOVF,
+	        MPU_REGION_MAX
+	};
+	```
+	- The ordering of MPU regions is kept as above taking into consideration below rule:
 		- For overlapping address range between MPU regions, attributes of higher MPU region number will override those set by lower numbered MPU region.
-	- Your new MPU region should be added in the enum above.
+	- The new MPU region(nregion_xx) other than the existing usages of board specific, common binary, app binary & stack overflow should be added in the structure above.
+	- In order to support few regions for the board specific purpose, below API is provided to reserve the first *num* number of regions:
+	```c
+	void mpu_set_nregion_board_specific(uint8_t num);
+	```
+	Above function shall be called during board initialization after the board specific MPU regions have been configured. The TizenRT MPU regions are configured after *num* number of board MPU regions.
```

Comment:
Let us move this point about board specific region to the new section before this section.